### PR TITLE
Refactor progress into shared search class

### PR DESCRIPTION
**Pull request by petersistrom**

### CODE REVIEW

File: classes/file_search.php [hunk 1]
```diff
@@ -335,16 +307,7 @@ public static function combine_shard_output(files $parent): void {
...
-            $fileinfo = [
-                'contextid' => \context_system::instance()->id,
-                'component' => 'tool_advancedreplace',
-                'filearea'  => 'files',
-                'itemid'    => $parent->get('id'),
-                'filepath'  => '/',
-                'filename'  => $parent->get_filename(),
-            ];
-            $fs->create_file_from_pathname($fileinfo, $outputpath);
+            $parent->save_pluginfile($output);
```
Comment (bwalkerl) [inline classes/file_search.php]: The param in save_pluginfile needs to be updated to $outputpath.

**Review by bwalkerl — COMMENTED** [review]
Thanks @petersistrom I found 1 issue with the rebase. 

Outside of that, when testing cli the --help find_in_files I noticed the docs list openzips instead of open-zips. Not entirely related but since this contains some CLI fixes we may as well fix that as well.

File: classes/file_search.php (hunk 1 above)
Comment (petersistrom) [reply classes/file_search.php]: Thanks @bwalkerl updated with both those fixes

File: cli/find_in_files.php [hunk 2]
```diff
@@ -51,7 +51,7 @@
 --filenames=filename            Cooma-separated list of file names to be searched.
 --skip-filenames=filename       Comma-separated list of file names to be omitted from the search.
 --skip-areas=areaname           Areas to skip. Separate multiple areas with a comma.
---openzips                      Open zip files, and search the files inside.
+--open-zips                      Open zip files, and search the files inside.
```
Comment (bwalkerl) [inline cli/find_in_files.php]: Small indenting inconsistency

Comment (petersistrom) [reply cli/find_in_files.php]: Fixed, along with a couple of other typos while I was there

**Pull request merged**